src/Contracts/ServiceInterface.php

```diff
@@ -11,66 +11,62 @@
+     * empty string.
      *
      * @param string $pageType One of self::PAGE_* which page is the customer on
-     *
-     * @return string|null
+     * @return string
      */
-    public function trackingCode(string $pageType): ?string;
+    public function trackingCode(string $pageType): string;
 
     /**
-     * Validate a request, this is the main entry point. If the returned response is async, @see updateRequest
+     * Validate a request. This is the main entry point. If the returned
+     * response is pending, @see updateRequest
      *
      * @param Request $request
-     *
      * @return \Omnifraud\Contracts\ResponseInterface
      * @throws \Omnifraud\Request\RequestException
      */
     public function validateRequest(Request $request): ResponseInterface;
 
     /**
-     * Update a previous async request, might still return an async response if the service needs more time to process
+     * Update a previous request. Useful for updating pending responses, or
+     * responses that were manually modified through the services web interface.
      *
      * @param Request $request Most services only require the UID
-     *
      * @return \Omnifraud\Contracts\ResponseInterface
      * @throws \Omnifraud\Request\RequestException
      */
     public function updateRequest(Request $request): ResponseInterface;
 
     /**
-     * If the service provides a web interface, this should return the URL at which the request can be viewed
+     * Generates a link to the services web view of a request/response. If the
+     * service does not provide dedicated web views, null is returned.
      *
      * @param string $requestUid
-     *
      * @return string|null
      */
     public function getRequestExternalLink(string $requestUid): ?string;
 
     // Those methods can be left empty if the service does not support them
 
     /**
-     * Send a refused transaction to the fraud prevention service if it was refused by the payment processor
-     *
-     * Can be left empty if not supported
+     * Send a refused transaction to the fraud prevention service if it was
+     * refused by the payment processor.
      *
      * @param Request $request
-     *
      * @return void
      * @throws \Omnifraud\Request\RequestException
      */
     public function logRefusedRequest(Request $request): void;
 
     /**
-     * Cancel a fraud review if it was refunded and/or manually refused.
-     * Some service will refund the review fee for those.
-     *
-     * Can be left empty if not supported
+     * Cancel a fraud review if it was refunded and/or manually refused. Some
+     * service will refund the review fee for those.
      *
-     * @param string $requestUid
+     * @param Request $request Most services only require the UID
      * @throws \Omnifraud\Request\RequestException
      */
-    public function cancelRequest(string $requestUid): void;
+    public function cancelRequest(Request $request): void;
```

Comment:
For consistency with `updateRequest`, I changed the type from a string to `Request`. If there is ever a 3rd party service that requires more than just the UID, the interface of every implementation will have to break (either by adding more properties to the interface, or changing from string to something else).
This seems like a safe, future-proof approach.